### ISSUE TITLE
CrashedLastRun is false if app gets out of memory termination

### ISSUE THREAD

**Issue by andrey-vishnitskiy-hypemasters** [getsentry/sentry-unity#1641]
### Environment

How do you use Sentry?
Sentry SaaS (sentry.io)

Which version of the SDK?
1.7.1

How did you install the package? (Git-URL, Assetstore)
git url

Which version of Unity?
2021.3.37

Is this happening in Unity (editor) or on a player like Android, iOS, Windows?
iOS, Android

### Steps to Reproduce

1. Open app
2. Allocate 2gb memory by creating textures
3. Game crashes
4. Reopen game
5. Cache SentryUnityOptions at SentryRuntimeConfiguration.cs
6. Call `options.CrashedLastRun?.Invoke()` after 10 sec

### Expected Result

Have options.CrashedLastRun true because last session has a crash due out of memory

### Actual Result

Have options.CrashedLastRun false

### Any logs or screenshots
[logs filtered by crash word.log](https://github.com/getsentry/sentry-unity/files/15235105/logs.filtered.by.crash.word.log)

**Comment by bitsandfoxes** [comment]
I get where you are coming from but the mechanism surrounding an Out Of Memor Crash is different and based on a heuristic and not a crash captured by the native SDKs.
But I think that's a fair point, if the native SDKs assume that the app crashed due to an OOM event it should consider the the previous run as crashed. I'll bring this up with the team.

**Comment by andrey-vishnitskiy-hypemasters** [comment]
Hi @bitsandfoxes, do you have any ETA for this topic?

**Comment by bitsandfoxes** [comment]
Sorry @andrey-vishnitskiy-hypemasters, unfortunately, we don't have an ETA yet.

**Comment by bitsandfoxes** [comment]
Confirming, that as long as the native SDKs are reporting these terminations, the `isCrashedLastRun` should return `true`.

**Comment by bruno-garcia** [comment]
> Confirming, that as long as the native SDKs are reporting these terminations, the isCrashedLastRun should return true.

@philipphofmann does the cocoa SDK report `true` on OOM?

**Comment by philipphofmann** [comment]
No, it doesn't. It only returns true for `crashedLastRun` if SentryCrash detected a crash. I'm open to changing that if you think it makes sense.

https://github.com/getsentry/sentry-cocoa/blob/5c3fa88fefada71a9fd8c567cffd03f875dd94c7/Sources/Sentry/SentrySDK.m#L449-L452

**Comment by bitsandfoxes** [comment]
> I'm open to changing that if you think it makes sense.

Conceptionally, the game did not close down "on its own volution" which - in my opinion - constitutes as a "crash" from the users perspective. Especially, since the SDK now surfaces that `CrashedLastRun`, there will be issues with false positives.

**Comment by philipphofmann** [comment]
I'm unsure if you think we should change `crashedLastRun` to include watchdog terminations and maybe also fatal app hangs or not?

**Comment by philipphofmann** [comment]
I created the issue: https://github.com/getsentry/sentry-cocoa/issues/4959.
